### PR TITLE
config: enable tichi for tidb

### PR DESCRIPTION
**Pull request by hi-rustin**
close https://github.com/ti-community-infra/configs/issues/147

### CODE REVIEW

**Comment by hi-rustin** [comment]
/hold

Waiting enable label plugin.

**Comment by hi-rustin** [comment]
/hold 

Waiting for TiDB team approve.

**Comment by hi-rustin** [comment]
/assign @Mini256 

 Please double-check all configurations, otherwise it will affect tidb collaboration.

File: prow/config/config.yaml [hunk 1]
```diff
@@ -129,6 +129,63 @@ branch-protection:
...
+              protect: true
+              required_status_checks:
+                contexts:
+                  - "idc-jenkins-ci-tidb/build"
+                  - "idc-jenkins-ci-tidb/check_dev"
+                  - "idc-jenkins-ci-tidb/check_dev_2"
+                  - "idc-jenkins-ci-tidb/check_release_note"
+                  - "license/cla"
+                strict: true
+            release-3.0:
```
Comment (zz-jason) [inline prow/config/config.yaml:156]: miss release 3.1

File: prow/config/config.yaml [hunk 2]
```diff
@@ -129,6 +129,63 @@ branch-protection:
...
+              protect: true
+              required_status_checks:
+                contexts:
+                  - "idc-jenkins-ci-tidb/build"
+                  - "idc-jenkins-ci-tidb/check_dev"
+                  - "idc-jenkins-ci-tidb/check_dev_2"
+                  - "idc-jenkins-ci-tidb/check_release_note"
+                  - "license/cla"
+                strict: true
+            release-5.0-rc:
```
Comment (zz-jason) [inline prow/config/config.yaml:178]: we'll need release 5.0 soon

File: prow/config/config.yaml (hunk 1 above)
Comment (hi-rustin) [reply prow/config/config.yaml:156]: There doesn't seem to be a version 3.1 branch on tidb.

image

File: prow/config/config.yaml (hunk 2 above)
Comment (hi-rustin) [reply prow/config/config.yaml:178]: We'll configure it later. This configuration can be updated at any time, and the PR merge will automatically update to the bot k8s cluster to take effect.

File: prow/config/external_plugins_config.yaml [hunk 3]
```diff
@@ -210,6 +231,42 @@ ti-community-label:
...
+      - 'needs-cherry-pick-3.1'
+      - 'needs-cherry-pick-4.0'
+      - 'needs-cherry-pick-5.0-rc'
+      - 'proposal'
+      - 'release-note'
+      - 'require-LGT3'
+      - 'security'
+      - 'tools'
+    exclude_labels:
+      - status/can-merge
```
Comment (Mini256) [inline prow/config/external_plugins_config.yaml:269]: Should `rewarded` and `picked` be added to `exclude_labels`? And how about PCP` and `PCP-S1` labels?

**Review by zhouqiang-cl — APPROVED** [review]
LGTM

File: prow/config/config.yaml [hunk 4]
```diff
@@ -129,6 +129,63 @@ branch-protection:
                   - "unit-test"
                   - "license/cla"
                 strict: true
+        tidb:
+          branches:
+            master:
+              enforce_admins: false
```
Comment (zz-jason) [inline prow/config/config.yaml:135]: can this config forbid admin to merge PRs when the required checks are failed?

Comment (hi-rustin) [reply prow/config/config.yaml:135]: Yes, just change the value to true. Do we want to force it on admins as well?

File: prow/config/config.yaml [hunk 5]
```diff
@@ -420,3 +482,99 @@ tide:
...
+                  - "idc-jenkins-ci-tidb/build"
+                  - "idc-jenkins-ci-tidb/check_dev"
+                  - "idc-jenkins-ci-tidb/check_dev_2"
+                  - "idc-jenkins-ci-tidb/check_release_note"
+                  - "license/cla"
+                  - "idc-jenkins-ci-tidb/check_title"
+                  - "idc-jenkins-ci-tidb/common-test"
+                  - "idc-jenkins-ci-tidb/e2e-test"
+                  - "idc-jenkins-ci-tidb/integration-common-test"
+                  - "idc-jenkins-ci-tidb/integration-compatibility-test"
```
Comment (zz-jason) [inline prow/config/config.yaml:518]: miss `integration-copr-test` and `integration-ddl-test`

File: prow/config/config.yaml [hunk 6]
```diff
@@ -420,3 +482,99 @@ tide:
...
```
Comment (zz-jason) [inline prow/config/config.yaml:536]: ditto

File: prow/config/config.yaml [hunk 7]
```diff
@@ -420,3 +482,99 @@ tide:
...
+                  - "idc-jenkins-ci-tidb/integration-copr-test"
```
Comment (zz-jason) [inline prow/config/config.yaml:555]: miss `integration-ddl-test`

File: prow/config/config.yaml [hunk 8]
```diff
@@ -420,3 +482,99 @@ tide:
...
```
Comment (zz-jason) [inline prow/config/config.yaml:574]: miss `integration-ddl-test`

File: prow/config/config.yaml (hunk 4 above)
Comment (hi-rustin) [reply prow/config/config.yaml]: Enabled.

**Comment by hi-rustin** [comment]
/lgtm cancel

@Mini256  Please take a look again.

File: prow/config/external_plugins_config.yaml [hunk 9]
```diff
@@ -210,6 +231,42 @@ ti-community-label:
...
+      - closed
+      - component
+      - difficulty
+      - epic
+      - feature
+      - severity
+      - sig
+      - status
+      - type
+    additional_labels:
```
Comment (zz-jason) [inline prow/config/external_plugins_config.yaml:247]: should we list all labels to here?

File: prow/config/config.yaml (hunk 5 above)
Comment (hi-rustin) [reply prow/config/config.yaml:518]: No these tests on this branch.

File: prow/config/config.yaml (hunk 6 above)
Comment (hi-rustin) [reply prow/config/config.yaml:536]: ditto

File: prow/config/config.yaml (hunk 7 above)
Comment (hi-rustin) [reply prow/config/config.yaml:555]: ditto

File: prow/config/config.yaml (hunk 8 above)
Comment (hi-rustin) [reply prow/config/config.yaml:574]: ditto

File: prow/config/external_plugins_config.yaml (hunk 9 above)
Comment (hi-rustin) [reply prow/config/external_plugins_config.yaml:247]: No, you can refer to https://book.prow.tidb.io/#/plugins/label.

The lables that need to be listed here are those that cannot be classified.

File: prow/config/config.yaml (hunk 5 above)
Comment (hi-rustin) [reply prow/config/config.yaml:518]: image

Comment (hi-rustin) [reply prow/config/config.yaml:518]: See: https://github.com/pingcap/tidb/pull/22838

**Comment by hi-rustin** [comment]
@zz-jason  Any other issues?

**Review by Mini256 — APPROVED** [review]

**Review by zz-jason — APPROVED** [review]
LGTM

**Comment by hi-rustin** [comment]
@zhouqiang-cl @zz-jason Thanks for your support.

**Pull request merged**